File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/bot/util/reply/CountReply.java
package lunalauf.rms.utilities.network.bot.util.reply;

import org.telegram.telegrambots.meta.api.methods.updatingmessages.EditMessageReplyMarkup;
import org.telegram.telegrambots.meta.api.objects.Message;
import org.telegram.telegrambots.meta.exceptions.TelegramApiException;

public class CountReply extends InlineReply {

	public CountReply(long chatId, ReplyContainer wrapper, Message message) {
		super(chatId, wrapper, message);
	}

	@Override
	public void runReplacedAction() throws TelegramApiException {
		EditMessageReplyMarkup newMarkup = new EditMessageReplyMarkup();
		newMarkup.setChatId(Long.toString(this.chatId));
		newMarkup.setMessageId(message.getMessageId());
		wrapper.getBot().execute(newMarkup);
	}

}

File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/bot/util/reply/CountReply.kt
package lunalauf.rms.utilities.network.bot.util.reply

import org.telegram.telegrambots.meta.api.methods.updatingmessages.EditMessageReplyMarkup
import org.telegram.telegrambots.meta.api.objects.Message
import org.telegram.telegrambots.meta.exceptions.TelegramApiException

class CountReply(
    chatId: Long,
    wrapper: ReplyContainer,
    message: Message
) : InlineReply(chatId, wrapper, message) {
    @Throws(TelegramApiException::class)
    override fun runReplacedAction() {
        val newMarkup = EditMessageReplyMarkup()
        newMarkup.chatId = chatId.toString()
        newMarkup.messageId = message.messageId
        wrapper.bot.execute(newMarkup)
    }
}
